src/Umbraco.Web/PropertyEditors/RichTextPropertyEditor.cs

```diff
@@ -131,15 +136,19 @@ public override object FromEditor(Core.Models.Editors.ContentPropertyData editor
             /// </summary>
             /// <param name="value"></param>
             /// <returns></returns>
-            public IEnumerable<Udi> GetReferences(object value)
+            public IEnumerable<UmbracoEntityReference> GetReferences(object value)
             {
                 var asString = value == null ? string.Empty : value is string str ? str : value.ToString();
 
+                //TODO: FindUdisFromDataAttributes will return UDIs of any type found, typically these will always be "media" but
```

Comment:
We basically needs a mapping from `Udi.EntityType` into the correct `Constants.Conventions.RelationTypes.*` Maybe with a fallback to something defined outside.

Reply:
What is the reason to explicit set the `RelationTypes` instead of using the info from the `udi`? And only take the UDI as input parameter in the ctor of  `UmbracoEntityReference`